### DISCUSSION TITLE
Remove Turbo's cache for a given repo

### DISCUSSION THREAD

**Discussion by WillsterJohnson** [vercel/turborepo#5270] [Help] [answered]
I'm using PNPM and Turbo in my monorepo.

Something went wrong with my production build, so I run `git clean -xdf` to get to the same place the production server is at. I've added everything which isn't source code or configs to `.gitignore` (including specifically `**/.turbo`), so `node_modules`, `dist` folders, etc., are removed.

I run `pnpm i` and `turbo build` to see what happens.

It fails for an entirely different reason to my production server. The build of one of my packages fails because it cannot find one of it's workspace dependencies. 

I check, sure enough it isn't there. There is no `dist/` folder.

I instruct Turbo to build that package only using `--filter`, and it tells me `cache hit (outputs already on disk), replaying logs`.

So I have two questions.

The first is why does Turbo think the outputs are there when they do not exist, the second is how can I tell Turbo to discard it's cache on my local machine?

I know I can use `--force`, but that's guaranteed to get missed at some point and either lead to a false fix or a false issue.

**Comment by mehulkar** [comment, marked as answer]
This seems like a bug, but it's likely that it's been fixed. The short story is that we have a background daemon that watches for file changes. There was a bug in this daemon where full deletions of folders sometimes didn't get registered, causing Turbo to _think_ the output is still there, when it wasn't. If you're still seeing this bug, please file an issue!
